mobile_back_apple/cpp/backend_coreml/coreml_util.swift

```diff
@@ -110,12 +170,21 @@ public class CoreMLExecutor: NSObject {
   private var inputNames: [String] = []
   private var outputNames: [String] = []
   private var outputProvider: MLBatchProvider?
+  private var accelerator: MLCommonsComputationUnit = MLCommonsComputationUnit(unitName: .all)
 
   @objc
-  public init(modelPath: UnsafePointer<CChar>, batchSize: Int) throws {
+  public init(
+    modelPath: UnsafePointer<CChar>, batchSize: Int, acceleratorName: UnsafePointer<CChar>
+  ) throws {
     pthread_set_qos_class_self_np(QOS_CLASS_USER_INITIATED, 0)
     let config = MLModelConfiguration()
-    config.computeUnits = MLComputeUnits.all
+
+    let unitName = String(validatingUTF8: acceleratorName)
+
+    if let name = unitName, let computationUnit = ComputationUnitNameEnum(rawValue: name) {
+      accelerator = MLCommonsComputationUnit(unitName: computationUnit)
+      config.computeUnits = accelerator.computeUnits
```

Comment:
What happens if `.cpuAndNeuralEngine` is selected / configured, but the device doesn't have an ANE?

Reply:
The option: .cpuAndNeuralEngine is first available from the IOS 16. So with this all devices have the ane. And the operating system decides on which chip runs the calculation. I think even if it not available if will run on cpu, the same way as .all option is working.

Reply:
iPhone 8 and some older models can still run on iOS 16 [1] and don't have ANE.
And `mlperf_backend_accelerator_name()` should report the actually used accelerator.

[1] https://support.apple.com/en-gb/guide/iphone/iphe3fa5df43/ios

Reply:
@freedom to check it on iPhone 8+ to see if it crashed or not.

Reply:
Confirm that when `cpu and ane` was chosen on iPhone 8+ (a device with ANE but not exposed to developers), CPU was used to run benchmark items. NO, it didn't crash.